### PR TITLE
Add controller landing page

### PR DESCRIPTION
**Pull request by jfim**
Add a simple landing page to the controller instead of returning a
404 not found when accessing / on the controller.

### CODE REVIEW

File: pinot-controller/src/main/resources/landing/index.html [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      <p><a class="btn btn-default" href="/query" role="button">Query console &raquo;</a></p>
+    </div>
+    <div class="col-md-4">
+      <h2>Pinot REST API</h2>
+      <p>An interactive documentation for the Pinot REST API with a swanky Swagger-generated page. Keep in mind that your actions here will directly affect the cluster, so be mindful of other tenants.</p>
+      <p><a class="btn btn-default" href="/help" role="button">REST API &raquo;</a></p>
+    </div>
+<!--    <div class="col-md-4">
+      <h2>Heading</h2>
+      <p>Donec sed odio dui. Cras justo odio, dapibus ac facilisis in, egestas eget quam. Vestibulum id ligula porta felis euismod semper. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus.</p>
```
Comment (kishoreg) [inline pinot-controller/src/main/resources/landing/index.html]: what is this text?

File: pinot-controller/src/main/resources/landing/index.html [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+  <hr>
+
+  <footer>
+    <p>&copy; 2014-2016 LinkedIn Corp.</p>
+  </footer>
+</div> <!-- /container -->
+
+</body>
+</html>
```
Comment (kishoreg) [inline pinot-controller/src/main/resources/landing/index.html]: need a new line here

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/ControllerRestApplication.java [hunk 3]
```diff
@@ -153,6 +153,11 @@ public void handle(Request request, Response response) {
 
     final Redirector redirector = new Redirector(getContext(), "/swagger-ui/index.html?url=/api", Redirector.MODE_CLIENT_TEMPORARY);
     router.attach("/help", redirector);
+
+    final Directory landing = new Directory(getContext(), getClass().getClassLoader().getResource("landing").toString());
```
Comment (kishoreg) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/ControllerRestApplication.java]: better to check for resource being null? I know it cant be if classpath is set properly but hard to debug a NPE. I suggest adding a warning if the resource is not found and move on.

**Comment by kishoreg** [comment]
LGTM. please take care of the minor comments

File: pinot-controller/src/main/resources/landing/index.html [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      </form> -->
+    </div><!--/.navbar-collapse -->
+  </div>
+</nav>
+
+<!-- Main jumbotron for a primary marketing message or call to action -->
+<div class="jumbotron">
+  <div class="container">
+    <h1>Welcome to Pinot!</h1>
+    <p>You're currently connected to a Pinot controller, which handles the cluster-wide coordination for all Pinot nodes. Most humans tend to find the following features useful:</p>
```
Comment (antumbde) [inline pinot-controller/src/main/resources/landing/index.html]: suggest re-wording "Most humans tend to find the"

File: pinot-controller/src/main/resources/landing/index.html [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    </div>
+<!--    <div class="col-md-4">
+      <h2>Heading</h2>
+      <p>Donec sed odio dui. Cras justo odio, dapibus ac facilisis in, egestas eget quam. Vestibulum id ligula porta felis euismod semper. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus.</p>
+      <p><a class="btn btn-default" href="#" role="button">View details &raquo;</a></p>
+    </div> -->
+  </div>
+
+  <hr>
+
```
Comment (antumbde) [inline pinot-controller/src/main/resources/landing/index.html]: suggestion: add a link to the product page on github.
Project page : Github link
For any issues/comments: contact

File: pinot-controller/src/main/resources/landing/index.html [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      <h2>Query console</h2>
+      <p>A place to conveniently run queries on the Pinot cluster. This is meant to be used for debugging, production queries should be sent directly to the appropriate brokers, to avoid having the controller forward the query to the broker.</p>
+      <p><a class="btn btn-default" href="/query" role="button">Query console &raquo;</a></p>
+    </div>
+    <div class="col-md-4">
+      <h2>Pinot REST API</h2>
+      <p>An interactive documentation for the Pinot REST API with a swanky Swagger-generated page. Keep in mind that your actions here will directly affect the cluster, so be mindful of other tenants.</p>
+      <p><a class="btn btn-default" href="/help" role="button">REST API &raquo;</a></p>
+    </div>
+<!--    <div class="col-md-4">
```
Comment (antumbde) [inline pinot-controller/src/main/resources/landing/index.html]: remove

File: pinot-controller/src/main/resources/landing/index.html [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        <span class="sr-only">Toggle navigation</span>
+        <span class="icon-bar"></span>
+        <span class="icon-bar"></span>
+        <span class="icon-bar"></span>
+      </button>
+      <a class="navbar-brand" href="#">Pinot Controller</a>
+    </div>
+    <div id="navbar" class="navbar-collapse collapse">
+
+      <!-- <form class="navbar-form navbar-right">
```
Comment (antumbde) [inline pinot-controller/src/main/resources/landing/index.html]: remove

**Comment by antumbde** [comment]
mostly minor optional suggestions. Kishore's comment on handling NPE is important. lgtm after addressing that

File: pinot-controller/src/main/resources/landing/index.html (hunk 1 above)
Comment (jfim) [inline pinot-controller/src/main/resources/landing/index.html]: Copy-paste from bootstrap template 😛 

I'll remove it.

**Comment by jfim** [comment]
We should review what we want to put on the web UI when I come back. Right now it's a mishmash of quickly made pages, but it would be good to have a much better landing page than this, especially if we can add information about servers, tables, cluster health, etc.

**Pull request merged**